Epam/Java Basics_UpSkill Lab #1/Algorithmization/Decomposition_using_methods/src/com/androsor/decomp/HomeTask4.java

```diff
@@ -0,0 +1,67 @@
+package com.androsor.decomp;
+
+import java.util.Random;
+import java.util.Scanner;
+
+/**
+ * On the plane, n points are given by their coordinates. Write a method (methods) that determine between which of the pairs of points
+ *   the longest distance. Indication. Enter the coordinates of the points into the array.
+ */
+
+public class HomeTask4 {
+
+    public static void main(String[] args) {
+        int n; // Количество заданных точек.
+
+        Scanner sc = new Scanner(System.in);
+        System.out.print(" Введите количество заданных точек n = ");
+        n = sc.nextInt();
+
+        // Заносим точки в массив
+        double[] x = new double[n];
+        double[] y = new double[n];
+        enterArray(x, y);
+
+        System.out.println("Заданы точки с координатами:");
+        for (int i = 0; i < n; i++) {
+            System.out.printf("%1d-ая Точка с координатами (%.2f, %.2f)\n",i + 1, x[i], y[i]);
```

Comment:
I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [FS: Format string should use %n rather than n (VA_FORMAT_STRING_USES_NEWLINE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#fs-format-string-should-use-n-rather-than-n-va-format-string-uses-newline).
This format string includes a newline character (\n). In format strings, it is generally  preferable to use %n, which will produce the platform-specific line separator.